### PR TITLE
Allow input mappings to be edited

### LINKED ISSUE #7312: Quick Edit of Input Mapping
~~Inside Project Settings -> Input Map, you are unable to edit added action names. You would have to remove it and re-add it along with all the input mapping that was previously set to do so.~~

You should be able to double click input mapping in order to edit it, instead of having to delete it and readd it. It is a minor change, but would make it more user-friendly to a small degree :)

~~I believe it would be an improvement to allow users to edit those without having to remake the entire action.~~

_(Searched for this issue, but failed to find any opened issues)_

### PR DESCRIPTION
**Pull request by bojidar-bg**
Closes #7312 

Proof gif (before double-click started to work):
peek 2017-06-24 14-53

### CODE REVIEW

File: editor/project_settings.cpp [hunk 1]
```diff
@@ -1174,10 +1284,11 @@ void ProjectSettings::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("_action_adds"), &ProjectSettings::_action_adds);
 	ClassDB::bind_method(D_METHOD("_action_selected"), &ProjectSettings::_action_selected);
 	ClassDB::bind_method(D_METHOD("_action_edited"), &ProjectSettings::_action_edited);
+	ClassDB::bind_method(D_METHOD("_action_activated"), &ProjectSettings::_action_activated);
 	ClassDB::bind_method(D_METHOD("_action_button_pressed"), &ProjectSettings::_action_button_pressed);
 	ClassDB::bind_method(D_METHOD("_update_actions"), &ProjectSettings::_update_actions);
 	ClassDB::bind_method(D_METHOD("_wait_for_key"), &ProjectSettings::_wait_for_key);
-	ClassDB::bind_method(D_METHOD("_add_item"), &ProjectSettings::_add_item);
+	ClassDB::bind_method(D_METHOD("_add_item"), &ProjectSettings::_add_item, DEFVAL(NULL));
```
Comment (akien-mga) [inline editor/project_settings.cpp]: This fails with clang (OSX on Travis):
```
editor/project_settings.cpp:1291:75: error: conversion from 'long' to 'const Variant' is ambiguous
        ClassDB::bind_method(D_METHOD("_add_item"), &ProjectSettings::_add_item, DEFVAL(NULL));
                                                                                 ^~~~~~~~~~~~
```

**Review by akien-mga — CHANGES_REQUESTED** [review]

File: editor/project_settings.cpp (hunk 1 above)
Comment (bojidar-bg) [inline editor/project_settings.cpp]: Hm, maybe I should use DEFVAL(Variant())...

**Review by akien-mga — APPROVED** [review]

File: editor/project_settings.cpp [hunk 2]
```diff
@@ -362,7 +373,7 @@ void ProjectSettings::_wait_for_key(const Ref<InputEvent> &p_event) {
 	}
 }
 
-void ProjectSettings::_add_item(int p_item) {
+void ProjectSettings::_add_item(int p_item, Ref<InputEvent> p_exiting_event) {
```
Comment (Hinsbart) [inline editor/project_settings.cpp]: Why is the event exiting? Did you mean existing? ^^

Comment (bojidar-bg) [inline editor/project_settings.cpp]: Ooops. 😃

**Pull request merged**